var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -246,24 +246,17 @@ def check_fortran_compiler(self):
     # Also PETSc prefer to build it without internal superlu, likely due to
     # conflict in headers see
     # https://bitbucket.org/petsc/petsc/src/90564b43f6b05485163c147b464b5d6d28cde3ef/config/BuildSystem/config/packages/hypre.py
-    depends_on("hypre@:2.13+mpi~internal-superlu~int64", when="@:3.8+hypre+mpi~complex~int64")
-    depends_on("hypre@:2.13+mpi~internal-superlu+int64", when="@:3.8+hypre+mpi~complex+int64")
-    depends_on(
-        "hypre@2.14:2.18.2+mpi~internal-superlu~int64", when="@3.9:3.13+hypre+mpi~complex~int64"
-    )
-    depends_on(
-        "hypre@2.14:2.18.2+mpi~internal-superlu+int64", when="@3.9:3.13+hypre+mpi~complex+int64"
-    )
-    depends_on(
-        "hypre@2.14:2.22.0+mpi~internal-superlu~int64", when="@3.14:3.15+hypre+mpi~complex~int64"
-    )
-    depends_on(
-        "hypre@2.14:2.22.0+mpi~internal-superlu+int64", when="@3.14:3.15+hypre+mpi~complex+int64"
-    )
-    depends_on("hypre@2.14:+mpi~internal-superlu~int64", when="@3.16:+hypre+mpi~complex~int64")
-    depends_on("hypre@2.14:+mpi~internal-superlu+int64", when="@3.16:+hypre+mpi~complex+int64")
-    depends_on("hypre@develop+mpi~internal-superlu+int64", when="@main+hypre+mpi~complex+int64")
-    depends_on("hypre@develop+mpi~internal-superlu~int64", when="@main+hypre+mpi~complex~int64")
+    depends_on("hypre@:2.13~internal-superlu", when="@:3.8+hypre")
+    depends_on("hypre@2.14:2.18.2~internal-superlu", when="@3.9:3.13+hypre")
```

Comment:
Hm  - spack now lists petsc@3.11: - so perhaps some of these old version dependencies can be removed. I guess thats a different PR [for with fixes for all pkgs - not just hypre]

Reply:
I think we should modify one of them and retain as:
```
    depends_on("hypre@2.14:2.18.2~internal-superlu", when="@3.11:3.13+hypre")
```